typing/jkind.mli

```diff
@@ -186,16 +186,29 @@ module Violation : sig
   (** Prints a violation and the thing that had an unexpected jkind
       ([offender], which you supply an arbitrary printer for). *)
   val report_with_offender :
-    offender:(Format.formatter -> unit) -> Format.formatter -> t -> unit
+    ?jkind_of_type:(Types.type_expr -> Types.jkind_l) ->
```

Comment:
I'm concerned about this optional parameter. It's undocumented in the mli, and it's behavior in non-obvious. And accidentally omitting it seems plausible - I think it should be something that is explicitly thought about when writing client code. Maybe we should make a variant to represent it and make it required?